bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -195,7 +195,26 @@ private String getJsContent() {
                 headLibResource = Utils.resolveResource(resolver, headLibPath);
                 if (headLibResource == null) {
                     LOG.trace("No custom headlib for resource type {}.", resourceType);
-                    continue;
+
+                    // Get resource from the resource type.
+                    Resource coreResource = slingRequest.getResourceResolver().getResource(resourceType);
+                    if (coreResource == null) {
+                        LOG.debug("Can't access resource from resource type {}.", resourceType);
+                        continue;
+                    }
+
+                    // Get resource superType path from the resource type.
+                    String superTypePath = coreResource.getResourceSuperType();
+                    if (superTypePath == null) {
+                      LOG.trace("No resource superType from resource type {}.", resourceType);
+                        continue;
+                    }
+                    // Get headLibResource from resource superType.
+                    headLibResource = Utils.resolveResource(resolver, superTypePath + "/" + cfg.getHeadlibName() + "/" + JcrConstants.JCR_CONTENT);
```

Comment:
Lets move the `"/" + cfg.getHeadlibName() + "/" + JcrConstants.JCR_CONTENT` string to a variable outside this for loop so we aren't concatenating the same string for every resource type.

Reply:
Done!